railgun/hack/generators/controllers/networking/main.go

```diff
@@ -234,14 +267,66 @@ var controllerTemplate = `
 type {{.PackageAlias}}{{.Type}}Reconciler struct {
 	client.Client
 
+{{- if or .AcceptsIngressClassNameSpec .AcceptsIngressClassNameAnnotation}}
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	Proxy            proxy.Proxy
+	IngressClassName string
+{{- else}}
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 	Proxy  proxy.Proxy
+{{- end}}
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *{{.PackageAlias}}{{.Type}}Reconciler) SetupWithManager(mgr ctrl.Manager) error {
```

Comment:
IIUC, this implementation seems to ignore the `--process-classless-XXX` flags implemented in https://github.com/Kong/kubernetes-ingress-controller/pull/1262 following the discussion in https://github.com/Kong/kubernetes-ingress-controller/pull/1262#discussion_r622976024 which renders these settings ineffective.

We need to do one of the below:
- implement classless handling throughout Railgun (and integration test it)
- decide to drop classless resources altogether

Following the discussion cited above, 2.0 seems to me like a good moment to do the latter.

Reply:
So apparently answer to

> Does continued maintenance impose a burden beyond addition of the flags?

is yes? Arguments in favor of dropping it are then:

- The classless behavior both conflicted with other controllers and was confusing even without them (classless handled, but only if your ingress class is set to "kong"), which is why we made it not the default.
- The new default behavior was added [~8 months ago](https://github.com/Kong/kubernetes-ingress-controller/blob/main/CHANGELOG.md#0100---20200915), which is IMO a reasonably long period to be aware of the new, preferred default.
- The old classless behavior is incompatible with the Ingress V1 [default class spec](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class), which labels a single IngressClass the cluster default and assigns all Ingress resources without class information that class.
- It's more complexity to maintain, and that complexity will become worse if we try to somehow support both it and the proper V1 behavior in code (as mutually exclusive settings).

Reply:
I have created a follow-up item for this issue: https://github.com/Kong/kubernetes-ingress-controller/issues/1293 :bow:

Reply:
I have two arguments against doing this in a follow-up issue.

- From the eng perspective: because of the classless handling mismatch, this PR adds ingress class functionality that is effectively broken and violates the contract (to the extent of interpretation of classlessness). I am not comfortable with merging in a PR with a well known contract breach for several reasons: bus factor, potential change of priorities ahead, us forgetting to make a fix or making an ineffective one, etc.
- From the PM perspective: repeated behavior of closing an issue with an incomplete solution and creation of a follow-up issue to "fill in the gap" leads to a project where it's impossible to track progress, because we're moving the goalposts with each such created issue and end up "crawling" first over the initial goals, then over fixes for the gaps, then for fixes for their fixes, etc.